packages/open-next/src/converters/aws-cloudfront.ts

```diff
@@ -0,0 +1,223 @@
+
+  for (const [key, values] of Object.entries(rawHeaders)) {
+    for (const { value } of values) {
+      if (value) {
+        headers[key.toLowerCase()] = value;
+      }
+    }
+  }
+
+  return headers;
+}
+
+async function convertFromCloudFrontRequestEvent(
+  event: CloudFrontRequestEvent,
+): Promise<InternalEvent> {
+  const { method, uri, querystring, body, headers, clientIp } =
+    event.Records[0].cf.request;
+  return {
+    type: "core",
+    method,
+    rawPath: uri,
+    url: uri + (querystring ? `?${querystring}` : ""),
+    body: Buffer.from(
+      body?.data ?? "",
+      body?.encoding === "base64" ? "base64" : "utf8",
+    ),
+    headers: normalizeCloudFrontRequestEventHeaders(headers),
+    remoteAddress: clientIp,
+    query: convertToQuery(querystring),
+    cookies:
+      headers.cookie?.reduce((acc, cur) => {
+        const { key, value } = cur;
+        return { ...acc, [key ?? ""]: value };
+      }, {}) ?? {},
+  };
+}
+
+type MiddlewareEvent = {
+  type: "middleware";
+} & MiddlewareOutputEvent;
+
+function convertToCloudfrontHeaders(
+  headers: Record<string, OutgoingHttpHeader>,
+) {
+  const cloudfrontHeaders: CloudFrontHeaders = {};
+  Object.entries(headers)
+    .filter(
+      ([key]) =>
+        !CloudFrontBlacklistedHeaders.some((header) =>
+          typeof header === "string" ? header === key : header.test(key),
+        ),
+    )
+    .forEach(([key, value]) => {
+      if (key === "set-cookie") {
+        const cookies = parseCookies(`${value}`);
+        if (cookies) {
+          cloudfrontHeaders[key] = cookies.map((cookie) => ({
+            key,
+            value: cookie,
+          }));
+        }
+        return;
+      }
+      cloudfrontHeaders[key] = [
+        ...(cloudfrontHeaders[key] || []),
+        ...(Array.isArray(value)
+          ? value.map((v) => ({ key, value: v }))
+          : [{ key, value: value.toString() }]),
+      ];
+    });
+  return cloudfrontHeaders;
+}
+
+async function convertToCloudFrontRequestResult(
+  result: InternalResult | MiddlewareEvent,
+  originalRequest: CloudFrontRequestEvent,
+): Promise<CloudFrontRequestResult> {
+  let responseHeaders =
+    result.type === "middleware"
+      ? result.internalEvent.headers
+      : result.headers;
+  if (result.type === "middleware") {
+    const { method, clientIp, origin } = originalRequest.Records[0].cf.request;
+
+    // Handle external rewrite
+    if (result.isExternalRewrite) {
+      const serverResponse = createServerResponse(result.internalEvent, {});
+      await proxyRequest(result.internalEvent, serverResponse);
+      const externalResult = convertRes(serverResponse);
+      debug("externalResult", {
+        status: externalResult.statusCode.toString(),
+        statusDescription: "OK",
+        headers: convertToCloudfrontHeaders(externalResult.headers),
+        bodyEncoding: externalResult.isBase64Encoded ? "base64" : "text",
+        body: externalResult.body,
+      });
+      return {
```

Comment:
The `debug` and returned object are identical. Should use combine into 1 object. Probably not a huge deal but perhaps the duplicate `toString` and `convertToCfHeaders()` might add a small overhead.